### PR TITLE
Nc news 6

### PR DESCRIPTION
**Pull request by JackCollier**

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -93,6 +93,98 @@ describe("GET /api/articles/:article_id", () => {
...
+      .then(({ body }) => {
+        const { comments } = body;
+        expect(comments.length).toEqual(11);
+        comments.forEach((comment) => {
+          expect(comment).toHaveProperty("comment_id", expect.any(Number));
+          expect(comment).toHaveProperty("votes", expect.any(Number));
+          expect(comment).toHaveProperty("author", expect.any(String));
+          expect(comment).toHaveProperty("body", expect.any(String));
+          expect(comment).toHaveProperty("created_at", expect.any(String));
+        });
```
Comment (MKD01) [inline __tests__/api.test.js:159]: As mentioned above you can either add the article_id here or in the above test.

File: __tests__/api.test.js [hunk 2]
```diff
@@ -93,6 +93,98 @@ describe("GET /api/articles/:article_id", () => {
...
+  test("should return 200 status", () => {
+    return request(app).get("/api/articles/1/comments").expect(200);
+  });
+  test("should respond with an array of comments for the given article_id", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .then(({ body }) => {
+        expect(body).toHaveProperty("comments", expect.any(Array));
+      });
+  });
```
Comment (MKD01) [inline __tests__/api.test.js:146]: This test isnt testing everything that is mentioned in the description. This test is simply asserting that there is an array attached to the key "comments", but you are not testing that these comments belong to the given article_id, which is a good test and should be included here or in the test below (if you want to add this to the test below then this test description will need to be amended).

File: models/api-models.js [hunk 3]
```diff
@@ -20,3 +21,52 @@ exports.selectArticleById = (article_id) => {
...
+    .query(
+      `SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at ASC`,
+      [article_id]
+    )
+    .then(({ rows }) => {
+      if (!rows.length) {
+        const error = new Error("No comments found with the specified ID");
+        error.status = 404;
+        error.msg = "Comments not found";
+        throw error;
```
Comment (MKD01) [inline models/api-models.js:68]: This can also be done in a much simpler way using a promise.reject if you would like to take a look into that.

**Review by MKD01 — APPROVED** [review]
Great work so far, there are a few changes which I would like to see mentioned below, but something that is missing is a test for when an article id is valid and exists but simply has no comments. What should this test look like? Once you have added this please resubmit this pr before 👍

File: __tests__/api.test.js [hunk 4]
```diff
@@ -93,6 +93,106 @@ describe("GET /api/articles/:article_id", () => {
...
+          expect(comment).toHaveProperty("comment_id", expect.any(Number));
+          expect(comment).toHaveProperty("article_id", expect.any(Number));
+          expect(comment).toHaveProperty("votes", expect.any(Number));
+          expect(comment).toHaveProperty("author", expect.any(String));
+          expect(comment).toHaveProperty("body", expect.any(String));
+          expect(comment).toHaveProperty("created_at", expect.any(String));
+        });
+      });
+  });
+  test("should respond with an error message comments not found when no comments are found on a happy path", () => {
```
Comment (Hard-Luck) [inline __tests__/api.test.js:163]: Should this be an error? Is it normal for a post to potentially have no comments?

**Review by Hard-Luck — COMMENTED** [review]
Just the one change, have a little think about it then change it. If you're confident with the change and why, then once it's done you can merge, be careful with branches going forward :)

**Pull request merged**